chainio/clients/eigenpod/writer.go

```diff
@@ -84,4 +88,87 @@ func NewManagerWriter(
 	return newManagerChainWriter(manager, ethClient, logger, txMgr), nil
 }
 
-// TODO(madhur): Add methods to ChainWriter and ManagerChainWriter to interact with the contracts.
+func (w *ChainWriter) VerifyWithdrawalCredentials(
+	ctx context.Context,
+	beaconTimestamp uint64,
+	stateRootProof bindings.BeaconChainProofsStateRootProof,
+	validatorIndices []*big.Int,
+	validatorFieldsProofs [][]byte,
+	validatorFields [][][32]byte,
```

Comment:
might be nice to document what this is. Or define named types for self-documenting code.